plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/Plugin.java

```diff
@@ -0,0 +1,77 @@
+package io.jenkins.tools.pluginmanager.impl;
+
+import java.util.jar.JarFile;
+
+public class Plugin {
+    private String name;
+    private String version;
+    private String url;
+    private String archiveFileName;
+    private JarFile jarFile;
+    private boolean isPluginOptional;
+
+
+    public Plugin(String name, String version, String url) {
+        this.name = name;
+        this.version = version;
+        this.url = url;
+        this.archiveFileName = new StringBuffer(name).append(".jpi").toString();
```

Comment:
normally you want to avoid using StringBuffer, its synchronised, and when building strings you rarely need that, if you are dynamically building a string you would use a `StringBuilder` in most cases.

But for simple (especially static / no loops) cases creating a new string is either more or equally as efficient, and its more readable
```suggestion
        this.archiveFileName = name + ".jpi";
```

Reply:
We'll leave out threading until phase 3? ;)